src/kubelet/client/connector.go

```diff
@@ -0,0 +1,252 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/transport"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+)
+
+const (
+	apiProxyPath = "/api/v1/nodes/%s/proxy/"
+	httpSchema   = "http"
+	httpsSchema  = "https"
+)
+
+type Connector interface {
```

Comment:
It would be great to have some comments on exported types and functions. What does `Connector` do?